src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/JsonSourceGeneratorTests.cs

```diff
@@ -749,5 +749,65 @@ public class ClassWithObsolete
             CompilationHelper.CheckDiagnosticMessages(DiagnosticSeverity.Warning, generatorDiags, Array.Empty<(Location, string)>());
             CompilationHelper.CheckDiagnosticMessages(DiagnosticSeverity.Error, generatorDiags, Array.Empty<(Location, string)>());
         }
+
+        [Fact]
+        public static void NoErrorsWhenUsingReservedCSharpKeywords()
+        {
+            string source = @"using System;
+using System.Text.Json.Serialization;
+
+namespace Test
+{
+    [JsonSerializable(typeof(ClassWithPropertiesAndFieldsThatIreReservedKeywords))]
```

Comment:
nit: typo in `ClassWithPropertiesAndFieldsThatIreReservedKeywords`

Reply:
Fixed typo